### PR TITLE
Move from glog to klog

### PR DESCRIPTION
**Pull request by bison**
Moving from glog to klog to align with related projects.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -85,3 +85,7 @@ required = [
 [[constraint]]
   branch = "master"
   name = "github.com/golang/glog"
```
Comment (ingvagabund) [inline Gopkg.toml]: Does `klog` replaces `glog` completely? If so, this can be removed.

File: cmd/manager/main.go [hunk 2]
```diff
@@ -4,31 +4,44 @@ import (
...
+
+	// Sync the glog and klog flags.
+	flag.CommandLine.VisitAll(func(f1 *flag.Flag) {
+		f2 := klogFlags.Lookup(f1.Name)
+		if f2 != nil {
+			value := f1.Value.String()
+			f2.Value.Set(value)
+		}
+	})
+
```
Comment (ingvagabund) [inline cmd/manager/main.go]: Should be safe to just use:
```go
flag.Set("logtostderr", "true")
klog.InitFlags(nil)
flag.Parse()
```

Comment (bison) [reply cmd/manager/main.go]: This is from the example in the klog repo and how things like the cluster-api tools initialize things.

File: Gopkg.toml (hunk 1 above)
Comment (bison) [reply Gopkg.toml]: It's actually still used in some of the tests. Was going to fix that in another PR.

**Comment by ingvagabund** [comment]
Can you remove `test/e2e` directory as well (e.g in a separate PR)? We no longer use that code and it's still importing `github.com/golang/glog`.

**Comment by bison** [comment]
> Can you remove test/e2e directory as well (e.g in a separate PR)? We no longer use that code and it's still importing github.com/golang/glog.

Yep. Planned on a follow up PR for that.

File: cmd/manager/main.go (hunk 2 above)
Comment (ingvagabund) [reply cmd/manager/main.go]: No objection here, just noticed we no longer have to use that hack of iterating through the flags. So, it's worth simplifying it and testing if all the flags are still available and properly propagated to vendored glog.

File: test/e2e/operator_expectations.go [hunk 3]
```diff
@@ -13,7 +14,7 @@ import (
 	cvorm "github.com/openshift/cluster-version-operator/lib/resourcemerge"
 	kappsapi "k8s.io/api/apps/v1"
 	batchv1 "k8s.io/api/batch/v1"
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (michaelgugino) [inline test/e2e/operator_expectations.go]: maybe corev1 here?  Did you mean to add this change, seems a little out of place for this PR.

Comment (ingvagabund) [reply test/e2e/operator_expectations.go]: Automatically generated

**Comment by enxebre** [comment]
please let make sure's there's a follow up for getting rid of glog completely from the vendor and drop the VisitAll loop
/lgtm

File: test/e2e/operator_expectations.go (hunk 3 above)
Comment (bison) [reply test/e2e/operator_expectations.go]: Yeah, this is from running `goimports`.

**Pull request merged**